src/front/glsl/context.rs

```diff
@@ -579,8 +579,165 @@ impl Context {
                 let right_inner = self.typifier.get(right, &parser.module.types);
 
                 match (left_inner, right_inner) {
-                    (&TypeInner::Vector { .. }, &TypeInner::Vector { .. })
-                    | (&TypeInner::Matrix { .. }, &TypeInner::Matrix { .. }) => match op {
+                    (
+                        &TypeInner::Matrix {
+                            columns: left_columns,
+                            rows: left_rows,
+                            width: left_width,
+                        },
+                        &TypeInner::Matrix {
+                            columns: right_columns,
+                            rows: right_rows,
+                            width: right_width,
+                        },
+                    ) => {
+                        // Check that the two arguments have the same dimensions
+                        if left_columns != right_columns
+                            || left_rows != right_rows
+                            || left_width != right_width
+                        {
+                            parser.errors.push(Error {
```

Comment:
is this check necessary? general validation would produce a very similar error, I think

Reply:
Checking the width and the rows is not necessary, but the columns need to be checked since otherwise code that uses matrices with different columns would either fail on validation or truncate depending on the order.